Quandl.Shared.Modules/Excel/DummyHostService.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Quandl.Shared.Excel
+{
+
+    /// <summary>
+    /// Required by class factory. does nothing.
+    /// </summary>
+    class DummyHostService : IHostService
+    {
+        public void SetStatusBar(string message)
+        {
+            // do nothing
+        }
+    }
+}
```

Comment:
it is used as fallback in case initialization of class factory fails for whatever reason (or not implemented at all, for example in a unit test module), to avoid unnecessary opaque `NullReferenceException`. It is not used during normal operations.